File: common/src/debug/java/de/nicidienase/chaosflix/ChaosflixApplication.java
package de.nicidienase.chaosflix;

import android.app.Application;
import android.content.Context;
import android.support.multidex.MultiDex;

import com.orm.SugarContext;

import org.jetbrains.annotations.NotNull;

/**
 * Created by felix on 18.03.17.
 */

public class ChaosflixApplication extends Application {
	private static Context APPLICATION_CONTEXT;

	@Override
	public void onCreate() {
		super.onCreate();
		SugarContext.init(this);
		APPLICATION_CONTEXT = this;
	}

	@Override
	public void onTerminate() {
		SugarContext.terminate();
		super.onTerminate();
	}

	@Override
	protected void attachBaseContext(Context base) {
		MultiDex.install(base);
		super.attachBaseContext(base);
	}

	@NotNull
	public static Context getContext() {
		return APPLICATION_CONTEXT;
	}
}

File: common/src/debug/java/de/nicidienase/chaosflix/ChaosflixApplication.kt
package de.nicidienase.chaosflix

import android.app.Application
import com.facebook.stetho.Stetho
import com.squareup.leakcanary.LeakCanary

class ChaosflixApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        Stetho.initializeWithDefaults(this)
        if (LeakCanary.isInAnalyzerProcess(this)) {
            // This process is dedicated to LeakCanary for heap analysis.
            // You should not init your app in this process.
            return
        }
//        LeakCanary.install(this);
    }
}
